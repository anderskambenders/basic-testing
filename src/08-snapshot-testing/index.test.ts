import { generateLinkedList } from './index';

const values1 = ['node1', 'node2'];
const values2 = ['node1', 'node2', 'node3'];

describe('generateLinkedList', () => {
  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', () => {
    const linkedList = generateLinkedList(values1);
    const expectedLinkedList = {
      value: 'node1',
      next: {
        value: 'node2',
        next: {
          value: null,
          next: null,
        },
      },
    };
    expect(linkedList).toStrictEqual(expectedLinkedList);
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    const linkedList = generateLinkedList(values2);
    expect(linkedList).toMatchSnapshot();
  });
});
