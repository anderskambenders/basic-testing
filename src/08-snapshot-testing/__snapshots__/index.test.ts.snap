// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": null,
        "value": null,
      },
      "value": "node3",
    },
    "value": "node2",
  },
  "value": "node1",
}
`;
